magento/magento2
# Improve attribute checking

On a store with a large number of attribute sets, a lot of repeated checking is done for the same attributes.

### Description
Instead of repeating the checks every time, keep track of the attributes that have already been checked and should be skipped (invisible). This we way we don't have to do the same check for the next attribute set.

### Fixed Issues (if relevant)

### Manual testing scenarios
1. Create a store with a large number of attribute sets, and add some invisible attributes in those sets
2. Upload a csv file with products to import, and click the Check Data button

With this change the checking phase will be much faster than without the change, without any functional change.

### Contribution checklist
